tests/unit/v1/test_charm_v1.py

```diff
@@ -0,0 +1,438 @@
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+#
+# Learn more about testing at: https://juju.is/docs/sdk/testing
+import unittest
+from pathlib import Path
+from unittest.mock import Mock, patch
+
+import yaml
+from charms.lego_base_k8s.v1.lego_client import AcmeClient
+from charms.tls_certificates_interface.v4.tls_certificates import (
+    Certificate,
+    CertificateSigningRequest,
+    ProviderCertificate,
+    RequirerCSR,
+)
+from ops import testing
+from ops.model import ActiveStatus, BlockedStatus
+from ops.testing import Harness
+from pylego import LEGOError, LEGOResponse
+from pylego.pylego import Metadata
+
+from tests.unit.v1.certificates_helpers import (
+    generate_ca,
+    generate_certificate,
+    generate_csr,
+    generate_private_key,
+)
+
+testing.SIMULATE_CAN_CONNECT = True  # type: ignore[attr-defined]
+test_cert = Path(__file__).parent / "test_lego.crt"
+TLS_LIB_PATH = "charms.tls_certificates_interface.v4.tls_certificates"
+CERT_TRANSFER_LIB_PATH = "charms.certificate_transfer_interface.v1.certificate_transfer"
+CERTIFICATES_RELATION_NAME = "certificates"
+CA_TRANSFER_RELATION_NAME = "send-ca-cert"
+
+
+class AcmeTestCharm(AcmeClient):
+    def __init__(self, *args):
+        """Use the AcmeClient library to manage events."""
+        super().__init__(*args, plugin="example")
+        self.valid_config = True
+
+    def _validate_plugin_config_options(self, plugin_config: dict[str, str]) -> str:
+        if not self.valid_config:
+            return "invalid plugin configuration"
+        return ""
+
+
+class TestCharmV1(unittest.TestCase):
```

Comment:
Since we are creating the test suite from scratch I would encourage to use scenario

Reply:
I didn't want to bundle that change here, I would prefer to do it in a different PR

Reply:
But why, these are all new tests. We are introducing harness here the same way we would be introducing scenario?

Reply:
I simply copy pasted the tests from v0 and changed the setups from config to secrets+config. I could still do it here if we want to.

Reply:
nevermind i just did it